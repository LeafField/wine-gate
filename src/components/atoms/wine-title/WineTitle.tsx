import React, { FC } from "react";
import Link from "next/link";

type Props = {
  title: string;
  id: string;
};

const WineTitle: FC<Props> = ({ id, title }) => {
  return (
    <Link
      href={`/article/${id}`}
      className="block border border-gray px-2 py-4 transition-colors duration-300 hover:border-blue"
    >
      <div className="line-clamp-2 h-full w-full text-heading3">{title}</div>
    </Link>
  );
};

export default WineTitle;
