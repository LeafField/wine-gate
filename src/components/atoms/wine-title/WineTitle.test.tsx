import { screen, render } from "@testing-library/react";
import WineTitle from "./WineTitle";

describe("WineTitleの単体テスト", () => {
  test("WineTitleコンポーネントが存在することを確認する", () => {
    const { container } = render(<WineTitle id="1" title="ワインのタイトル" />);
    expect(screen.getByText("ワインのタイトル")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
