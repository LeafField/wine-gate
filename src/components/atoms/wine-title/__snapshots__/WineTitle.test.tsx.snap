// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WineTitleの単体テスト WineTitleコンポーネントが存在することを確認する 1`] = `
<div>
  <a
    class="block border border-gray px-2 py-4 transition-colors duration-300 hover:border-blue"
    href="/article/1"
  >
    <div
      class="line-clamp-2 h-full w-full text-heading3"
    >
      ワインのタイトル
    </div>
  </a>
</div>
`;
